client/allocrunner/alloc_runner.go

```diff
@@ -423,6 +484,34 @@ func (ar *allocRunner) Update(update *structs.Allocation) {
 		ar.logger.Error("error running update hooks", "error", err)
 	}
 
+	// If alloc is being terminated, kill all tasks, leader first
+	if stopping {
```

Comment:
Should we run ar.update hooks if it is stopping?

Reply:
I wondered this... Is there a chance information that needs to be processed is sent in the same alloc update as termination? For example could a canary tag be added or removed?

If so I _think_ we should call all update hooks, but even then it seems a little strange...